docs/content/focusTrap.mdx

```diff
@@ -0,0 +1,98 @@
+* External focus changes: If an external cause (e.g. mouse click, scripts, or accessibility software) results in an element outside the container to be focused, focus will immediately be redirected to the last-focused element that is inside the container.
+* Circular tab focus: Using the `TAB` key on the last focusable element within the container will result in the first focusable element within the container receiving focus. Similarly, `Shift+TAB` can be used to focus the last element from the first.
+* Global: Only one focus trap can be _active_ at a time. When a focus trap is enabled, if there is already an active focus trap, it becomes suspended and pushed onto a stack. Once the newly-active focus trap is disabled, the most recently-suspended trap will reactivate. Suspended focus traps can be disabled, causing them to be removed from the stack of suspended traps.
+
+### Demo
+
+To see a demo, deploy Storybook and find the `useFocusTrap` stories.
+
+### Usage
+
+```ts
+function showDialog() {
+  const dialog = document.getElementById("myDialog")
+  if (dialog instanceof HTMLElement) {
+    dialog.style.display = ""
+    return focusTrap(dialog)
+  }
+}
+function hideDialog(controller: AbortController) {
+  document.getElementById("myDialog")?.style.display = "none"
+  controller.abort()
+}
+const dialogController = showDialog()
+
+// later
+if (dialogController) {
+  hideDialog(controller)
+}
+```
+
+### API
+
+The `focusTrap` function takes the following arguments.
+
+| Name | Type | Default | Description |
+| :- | :- | :-: | :- |
+| container | `HTMLElement` | | When active, only elements within this container (along with the container itself) can be focused. |
+| initialFocus | `HTMLElement` | | Specifies the element which will receive focus when the focus trap is activated. Defaults to the first tabbable element inside the container. |
+| signal | `AbortSignal?` | `undefined` | Optional abort signal to control the focus trap. If one is not provided, an `AbortController` will be returned, which can be used to disable the focus trap. |
+
+#### Return value
+
+If the `signal` argument is omitted, `focusTrap()` will return an `AbortController`. This object has an `abort()` method, which can be called to disable the focus trap.
+
+### Best practices
+
+* Focus management is an important consideration for accessible applications. Sometimes poor focus management can make certain tasks impossible to complete for users not using a mouse. To learn more, read the [ARIA guidelines for keyboard focus](https://www.w3.org/TR/wai-aria-practices/#kbd_focus_discernable_predictable).
+* Only activate a focus trap if all UI outside of the trap container should be inert (non-interactive).
```

Comment:
This reminds me, initially we were planning on baking this into `Overlay` by default but it should probably be optional since we shouldn't be trapping focus in things like ActionMenu or Dropdown

Reply:
I don't think Overlay should trap focus - that should be left to Dialogs.